packages/tenant-process/src/services/tenantService.ts

```diff
@@ -354,15 +352,20 @@ export function tenantServiceBuilder(
           correlationId
         );
 
-      const [updatedTenant, tenantKindUpdatedEvent] =
+      const { updatedTenant, tenantKindHasBeenUpdated } =
         await reevaluateTenantKind({
           tenant: tenantWithNewAttribute,
-          version: targetTenant.metadata.version + 1,
           readModelService,
-          correlationId,
         });
 
-      if (tenantKindUpdatedEvent) {
+      if (tenantKindHasBeenUpdated) {
+        const tenantKindUpdatedEvent = toCreateEventTenantKindUpdated(
+          targetTenant.metadata.version + 1,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I would avoid solutions that ignore linting because it's very easy to miss something in future changes.
What if the `reevaluateTenantKind` returns the new kind? which is something I would expect based on its name.

Also, I don't think at this point that the function should return the updated tenant. The update can be done directly where the event is stored

Reply:
I link [here](https://github.com/pagopa/interop-be-monorepo/pull/407#discussion_r1693017575) a previous discussion on this.
I agree with @galales on changing function behavior.